ui/pages/device/index.js

```diff
@@ -33,6 +32,48 @@ export default function Device() {
     return false
   }
 
+  async function setCodeSegment(segment, pos) {
+    var codeCopy = code || ''
+    for (; codeCopy.length < 8; ) {
```

Comment:
What is this function doing? Adding spaces to a the code to populate the remaining boxes in the UI?

Reply:
yeah, it covers some really corner cases where you try to fill the last box first, etc.